File: Programowanie/Inne/Wprowadzenie do języka Ruby/Pliki źródłowe/klasy.rb

class Prostokat
  attr_accessor :wys
  attr_accessor :szer

  def initialize(wysokosc, szerokosc)
    @wys = wysokosc
    @szer = szerokosc
  end

  def pole
    @wys * @szer
  end

  def kwadrat?
    @wys == @szer
  end
end


prostokat = Prostokat.new(5, 10)
puts prostokat.pole
puts prostokat.kwadrat?

prostokat.wys = 10
puts prostokat.kwadrat?

File: Programowanie/C++/Wprowadzenie do języka/Pliki źródłowe/klasy.py
class Prostokat:

    def __init__(self, wysokosc, szerokosc):
        self.wys = wysokosc
        self.szer = szerokosc

    def pole(self):
        return self.wys * self.szer

    def czy_kwadrat(self):
        return self.wys == self.szer


prostokat = Prostokat(5, 10)
print(prostokat.pole())
print(prostokat.czy_kwadrat())

prostokat.wys = 10
print(prostokat.czy_kwadrat())
